client/src/components/CreateCard/cardContext.js

```diff
@@ -0,0 +1,176 @@
+import React, { createContext, useState, useContext } from "react";
+import moment from "moment";
+import { authFetch } from "../../AuthService";
+import { UserContext } from "../../userContext";
+import { handleError, handleSuccess } from "../../utils/handleAlerts";
+
+const CardContext = createContext();
+
+//To Handle state and comments globally
+const CardProvider = props => {
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [openCard, setOpenCard] = useState(false);
+  const [openTag, setOpenTag] = useState(false);
+  const [openDeadline, setOpenDeadline] = useState(false);
+  const [tag, setTag] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [columnName, setColumnName] = useState("");
+  const [error, setError] = useState("");
+  const [taskId, setTaskId] = useState("");
+  const [columnId, setColumnId] = useState("");
+
+  //get dashboard values from user context
+  const { value1 } = useContext(UserContext);
+  const [dashboard] = value1;
+  const dashboardId = dashboard._id;
+
+  const handleCurrentTask = (taskId, columnId) => {
+    const columnName = dashboard.columns[columnId].title;
+    if (!taskId) {
+      setTitle("");
+      setDescription("");
+      setDeadline("");
+      setTag("");
+      setTaskId("");
+    }
+    if (taskId) {
+      setTaskId(taskId);
+      fetchCard(taskId, columnId);
+    }
+    setColumnId(columnId);
+    setColumnName(columnName);
+    handleOpenCard();
+  };
+
+  const fetchCard = (taskId, columnId) => {
+    const task = dashboard.columns[columnId].tasks[taskId];
+    setTitle(task.title);
+    setDescription(task.description);
+    setTag(task.tag);
+    setDeadline(task.deadline);
+    task.deadline && setOpenDeadline(true);
+  };
+
+  const handleSubmit = () => {
+    if (!title) {
+      setError(true);
+    } else {
+      if (!taskId) {
+        const createTask = {
+          columnId,
+          dashboardId,
+          deadline,
+          title,
+          description,
+          tag
+        };
+        authFetch("/dashboards/task", {
+          method: "POST",
+          body: JSON.stringify(createTask)
+        })
+          .then(() => handleCloseCard())
+          .then(() => handleSuccess(`${title} has been saved!`))
+          .catch(err => {
+            handleError(err);
+          });
+      } else {
+        const updatedTask = {
+          columnId,
+          dashboardId,
+          taskId,
+          deadline,
+          title,
+          description,
+          tag
+        };
+        authFetch("/dashboards/task", {
```

Comment:
As you know, 
By following Aecio's code review, I changed all routes. 
Sorry for doing extra work, but please change all url for fetching data if you needed.